src/mame/marion/ms68k.cpp

```diff
@@ -0,0 +1,319 @@
+// license:BSD-3-Clause
+// copyright-holders:Chris Hanson
+/*
+ * ms68k.cpp - Marion Systems MS68K
+ *
+ * Documentation:
+ *   http://bitsavers.org/pdf/marionSystems/Marion_Systems_MS68K_Single_Board_Computer_Users_Manual.pdf
+ *
+ *  The Marion Systems MS68K was a single-board MC68000 system designed by Tom
+ *  Oberheim in 1985 with two serial ports, a parallel port, WD1772 floppy,
+ *  optional SCSI, and up to 512KB of RAM and ROM each onboard, plus an
+ *  expansion connector providing full access to the 68000 bus. It was used in
+ *  a number of university-level computer systems classes, and a variant of
+ *  Peter Stark’s SK*DOS was also available.
+ *
+ *  The default ROM contains a port of Tiny BASIC 1.2 at 0xF04900, and a
+ *  loader for it at 0xF04800. Thus while there is no direct command to start
+ *  it, a simple GF04800 will start Tiny BASIC with it.
+ *
+ * Specifications:
+ * - 8MHz MC68000 CPU
+ * - 512KB maximum onboard RAM
+ * - 128KB maximum onboard EPROM
+ * - MC68681 DUART
+ * - TTL-supported parallel port
+ * - NCR53C80 SCSI controller
+ * - Expansion bus
+ *
+ * TODO:
+ * - Expansion bus
+ *
+ */
+
+#include "emu.h"
+
+#include "bus/centronics/ctronics.h"
+#include "bus/nscsi/devices.h"
+#include "bus/rs232/rs232.h"
+#include "cpu/m68000/m68000.h"
+#include "imagedev/floppy.h"
+#include "machine/mc68681.h"
+#include "machine/ncr5380.h"
+#include "machine/wd_fdc.h"
+
+
+namespace {
+
+
+class ms68k_state : public driver_device
+{
+public:
+	ms68k_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_bootvect(*this, "bootvect")
+		, m_maincpu(*this, "maincpu")
+		, m_duart(*this, "duart")
+		, m_terminal(*this, "terminal")
+		, m_modem(*this, "modem")
+		, m_fdc(*this, "fdc")
+		, m_floppy(*this, "fdc:%u", 0)
+		, m_floppy_selected_drive(0)
+		, m_scsi(*this, "scsibus:7:ncr5380")
+		, m_scsibus(*this, "scsibus")
+		, m_printer_conn(*this, "prn")
+		, m_printer_out(*this, "prn_out")
+	{ }
+
+	void ms68k(machine_config &config) ATTR_COLD;
+
+	DECLARE_INPUT_CHANGED_MEMBER(nmi_button);
+
+protected:
+	virtual void machine_reset() override ATTR_COLD;
+
+private:
+	void mem_map(address_map &map) ATTR_COLD;
+
+	memory_view m_bootvect;
```

Comment:
Please put `mem_map` with the other member functions in the `private` section (either collect the member functions above or below the data members).